flaggr/src/main/java/com/comuto/flag/Flaggr.java

```diff
@@ -75,31 +75,43 @@ private Flaggr(Context context) {
      * @param configUrl url of json config.
      */
     public void loadConfig(String configUrl, String defaultFlagsFileName) {
+        loadConfig(configUrl, defaultFlagsFileName, null);
+    }
+
+    public void loadConfig(String configUrl, String defaultFlagsFileName, @Nullable final FlagsResultCallback callback){
         this.configUrl = configUrl;
         this.defaultFlagsFileName = defaultFlagsFileName;
-        downloadConfig(configUrl, defaultFlagsFileName);
+        downloadConfig(configUrl, defaultFlagsFileName, callback);
     }
 
     /**
      * Reload flags for json config.
      */
     public void reloadConfig() {
-        downloadConfig(this.configUrl, this.defaultFlagsFileName);
+        downloadConfig(this.configUrl, this.defaultFlagsFileName, null);
```

Comment:
for reloading wouldn't you want the caller to have their callback called again with the new config ?